### PR TITLE
[WIP] - V3 view documentation

### PR DESCRIPTION
**Pull request by vernondegoede**
My work on #2920 so far.  I'd still like to dive a little deeper into events and how they're triggered onto child views.

Also, I need to finish the merge of the LayoutView documentation into the View documentation. 

cc @scott-w

### CODE REVIEW

File: docs/marionette.view.md [hunk 1]
```diff
@@ -28,7 +24,7 @@ will provide features such as `onShow` callbacks, etc. Please see
 * [Organizing ui elements](#organizing-ui-elements)
 * [modelEvents and collectionEvents](#modelevents-and-collectionevents)
 
-## View render
+## Rendering
```
Comment (scott-w) [inline docs/marionette.view.md]: This should probably link to `#view-serializedata`

File: docs/marionette.view.md [hunk 2]
```diff
@@ -2,18 +2,14 @@
...
will be treated as a single item. It’s also possible to create Views that do not represent items, but just contain regions (which can be filled with View elements as well).

View extends directly from `Backbone.View`, which means you’re free to use Backbone’s methods in your Marionette Views.
 
 Additionally, interactions with Marionette.Region
 will provide features such as `onShow` callbacks, etc. Please see
 [the Region documentation](./marionette.region.md) for more information.
 
+Note: Since Marionette v3.x, `Marionette.View` will replace `Marionette.LayoutView` and `Marionette.ItemView`.
+
```
Comment (scott-w) [inline docs/marionette.view.md]: `onShow` is deprecated in v3 - `onRender` is now the recommended event to use.

**Comment by scott-w** [comment]
This is excellent work @vernondegoede thanks!

There's only two things that I think could improve this:

1. `onBeforeShow` and `onShow` are deprecated now - `onRender` can achieve the behaviour you want in `onBeforeShow`
2. I prefer to use the header structure # ## ### instead of bolded text for more clear semantic markup

Overall this is really good and I like how you're bringing in the basics of region management to make it easier for readers to get all the basic information they need.

**Comment by ahumphreys87** [comment]
nice work @vernondegoede! @scott-w do you think we can merge this and iterate on the LayoutView => ItemView merge docs? Or would it not make much sense to merge yet?

**Comment by scott-w** [comment]
I've had a read and it seems okay to merge this for now. There's still some work to be done, sections to remove, etc. but that can be fixed afterwards.

@vernondegoede Are you happy for me to merge this for now and you can just open a new PR later?

**Comment by vernondegoede** [comment]
@scott-w Yeah, you can merge this for now. As we discussed I still wanted to make some improvements, but didn't have time for that over the last two weeks :(

I'll open a new PR for my improvements when they're done.

**Comment by scott-w** [comment]
Awesome, cheers. I'll get this in now.

**Pull request merged**